packages/protocol/contracts/L1/TaikoL1.sol

```diff
@@ -89,13 +89,18 @@ contract TaikoL1 is EssentialContract, ITaikoL1, TaikoEvents, TaikoErrors {
             }
         }
 
+        require(_params.length == _txLists.length, "mismatched params length");
+
         TaikoData.Config memory config = getConfig();
         TaikoToken tko = TaikoToken(resolve(LibStrings.B_TAIKO_TOKEN, false));
 
-        (meta_, deposits_) = LibProposing.proposeBlock(state, tko, config, this, _params, _txList);
+        for (uint256 i = 0; i < _params.length; i++) {
+            (meta_, deposits_) =
+                LibProposing.proposeBlock(state, tko, config, this, _params[i], _txLists[i]);
 
-        if (LibUtils.shouldVerifyBlocks(config, meta_.id, true) && !state.slotB.provingPaused) {
-            LibVerifying.verifyBlocks(state, tko, config, this, config.maxBlocksToVerify);
+            if (LibUtils.shouldVerifyBlocks(config, meta_.id, true) && !state.slotB.provingPaused) {
+                LibVerifying.verifyBlocks(state, tko, config, this, config.maxBlocksToVerify);
+            }
```

Comment:
I think a bit more efficient to move this out of the loop and use `config.maxBlocksToVerify * _params.length` as max blocks to verify.

Reply:
will do!